salt/states/crmshmod.py

```diff
@@ -0,0 +1,281 @@
+'''
+State module to provide CRMSH (HA cluster) functionality to Salt
+
+.. versionadded:: pending
+
+:maintainer:    Xabier Arbulu Insausti <xarbulu@suse.com>
+:maturity:      alpha
+:depends:       crmsh
+:platform:      all
+
+:configuration: This module requires the crm python module and uses the
+    following defaults which may be overridden in the minion configuration:
+
+.. code-block:: yaml
+
+    #TODO: create some default configuration parameters if needed
+
+:usage:
+
+.. code-block:: yaml
+    cluster-init:
+      crm.initialized:
+        - name: 'hacluster'
+        - watchdog: '/dev/watchdog'
+        - interface: 'wlan0'
+        - unicast: False
+        - admin_ip: '192.168.1.50'
+        - sbd: False
+'''
+
+
+# Import python libs
+from __future__ import absolute_import, unicode_literals, print_function
+
+
+# Import salt libs
+from salt import exceptions
+from salt.ext import six
+
+
+__virtualname__ = 'crm'
+
+
+def __virtual__():  # pragma: no cover
+    '''
+    Only load if the hana module is in __salt__
+    '''
+    return 'crm.cluster_init' in __salt__
+
+
+def cluster_absent(
+        name='localhost',
+        quiet=None):
+    """
+    Machine is not running as a cluster node
+
+    quiet:
+        execute the command in quiet mode (no output)
+    """
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    if __salt__['crm.status']():
+        ret['result'] = True
+        ret['comment'] = 'Cluster is not running already'
```

Comment:
Maybe I didn't understand the logic here. If the result of `crm status` is true, doesn't it mean that the cluster is still running?
I would also change the message to:

```suggestion
        ret['comment'] = 'Cluster is already not running'
```

Reply:
Yeah, maybe it's not totally clear how it works...
Basically, __salt__['crm.status'} returns the returncode of the command crm status. So, if the code is 0, everything went fine, ofherwise some error happened.

We could improve that in future changes anyway